From e33acc217897861a0a535fa5ac05cec5f115e2fc Mon Sep 17 00:00:00 2001
From: Kiran Kumar Lokere <quic_klokere@quicinc.com>
Date: Mon, 12 Aug 2024 00:26:57 -0700
Subject: [PATCH] Add QCA vendor attribute to change P2P GO beacon interval

Add new QCA vendor attribute to configure the P2P GO beacon interval
dynamically while the BSS continues operating.

Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 src/common/qca-vendor.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/common/qca-vendor.h b/src/common/qca-vendor.h
index 5e7b67b5b..6bee6e565 100644
--- a/src/common/qca-vendor.h
+++ b/src/common/qca-vendor.h
@@ -3723,6 +3723,17 @@ enum qca_wlan_vendor_attr_config {
 	 */
 	QCA_WLAN_VENDOR_ATTR_CONFIG_FOLLOW_AP_PREFERENCE_FOR_CNDS_SELECT = 121,
 
+	/* 16-bit unsigned value to configure P2P GO beacon interval in TUs.
+	 * This attribute is used to update the P2P GO beacon interval
+	 * dynamically.
+	 *
+	 * Updating the beacon interval while the GO continues operating the BSS
+	 * will likely interoperability issues and is not recommended to be
+	 * used. All the values should be multiples of the minimum used value to
+	 * minimize risk of issues.
+	 */
+	QCA_WLAN_VENDOR_ATTR_CONFIG_P2P_GO_BEACON_INTERVAL = 122,
+
 	/* keep last */
 	QCA_WLAN_VENDOR_ATTR_CONFIG_AFTER_LAST,
 	QCA_WLAN_VENDOR_ATTR_CONFIG_MAX =
-- 
2.43.0

